internal/scheduler/nodedb/nodedb.go

```diff
@@ -768,7 +761,9 @@ func (nodeDb *NodeDb) selectNodeForJobWithUrgencyPreemption(
 			continue
 		}
 
-		if priority > req.Priority {
+		// Using pctx.ScheduledAtPriority instead of jctx.PodRequirements.Priority,
+		// since the pctx.ScheduledAtPriority may differ, e.g., in case of home-away scheduling.
+		if priority > pctx.ScheduledAtPriority {
```

Comment:
This was the actual issue to be fixed in this file. Other changes in this file is just refactoring for clarity.